docs/ext/commands/additional_info.rst

```diff
@@ -32,3 +32,30 @@ If your bot requires shard distribution across several machines, you should set
 This will prevent conflicts and race conditions. Discord API doesn't provide users with events related to application command updates,
 so it's impossible to keep the cache of multiple machines synced. Having only 1 machine with ``sync_commands`` set to ``True`` is enough
 because global registration of application commands doesn't depend on sharding.
+
+.. _why_params_and_injections_return_any:
+
+Why ``Param`` and ``Injection``-related functions return ``Any``?
+-----------------------------------------------------------------
+
+If your editor of choice supports type-checking, you might have been noticed that :func:`~ext.commands.Param`, :func:`~ext.commands.inject`,
+:func:`~ext.commands.injection` and :func:`~ext.commands.register_injection` return ``Any`` (in terms of typing), but at runtime
+return :class:`~ext.commands.Injection` (as mentioned in API reference for them). "Why?", you could think, and here's the reason for this.
+
+We'll explain this on example of ``Param`` usage (because this is the thing most users are familair with), but everything here applies to
+``Injection`` usage as well.
+
+A typical example of slash command might look like this: ::
+
+    @bot.slash_command(description="Replies with the given text!")
+    async def echo(
+        inter: disnake.ApplicationCommandInteraction,
+        text: str = commands.Param(description="Echo~")
+    ) -> None:
+        await inter.response.send_message(text)
+
+Here, you have two parameters in your command's function: ``inter``, an instance of :class:`disnake.ApplicationCommandInteraction`, and
+``text``, which is.. Noticed? You annotate ``text`` as ``str``, but at the same time, assign :class:`~ext.commands.ParamInfo` instance to it.
+That's the thing. When your editor type-checks your (any library's) code, it would normally complain if you tried to do the above, because
+you're trying to "convert" ``ParamInfo`` into ``str``, but since library have set ``Param``'s return type as ``Any``, type-checker "accepts"
+your code, because ``str`` (and any type) is the subset of ``Any``. Win-win!
```

Comment:
Needs some padding of some sort below this as it is currently occluded by the top and rtd buttons and cannot be scrolled any further. 
![image](https://user-images.githubusercontent.com/71233171/190725946-897b9ef0-b17e-4d36-85ae-e6c9bad608be.png)

Reply:
Done as of latest commit

Reply:
Unresolved due to the extra line being not necessary.